f8a_worker/process.py

```diff
@@ -356,8 +356,14 @@ def fetch_npm_artifact(name, version, target_dir):
         #      └── package.tgz
         # 3 directories, 6 files
         name_ver = name
+        invalidChars = set(string.punctuation.replace(".", ""))
```

Comment:
Aren't dashes also valid version characters in NPM? Like in `1.1.1-alpha1`. This is exactly my concern with this approach. We may easily refuse to analyze perfectly valid versions.

Idea: wouldn't be better to fetch list of all versions for the package in question and just check if the version given to us by user exists?

Reply:
Ok looking into sem ver definition only "." and "-" special characters are allowed.